language/tools/move-package/src/lib.rs

```diff
@@ -115,10 +120,14 @@ impl BuildConfig {
         model_config: ModelConfig,
     ) -> Result<GlobalEnv> {
         let resolved_graph = self.resolution_graph_for_package(path)?;
-        ModelBuilder::create(resolved_graph, model_config).build_model()
+        let mutx = PackageLock::lock();
+        let ret = ModelBuilder::create(resolved_graph, model_config).build_model();
+        mutx.unlock();
+        ret
     }
 
     pub fn resolution_graph_for_package(mut self, path: &Path) -> Result<ResolvedGraph> {
+        let mutx = PackageLock::lock();
```

Comment:
Not sure I immediately follow the granularity of this lock. I would intuitively expect it to go before 139 or 140, but I don't have a good understanding of which paths might be touched in other threads. Could be worth a comment.

Reply:
Good question, looking at it since this take ownership of `self` the lock only needs to be around 139/140, but I figured to be on the safe side I'd lock the whole body. I'll update

Reply:
I'm not worried about the perf of a coarser grained lock here, but I think placing them very precisely is good for understandability.